TestProject/output/BaseOutput.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace output;
+
+use options\Getopt;
+use cellularAutomat\Board;
+
+/**
+ * BaseOutput class which provides the functions for pluggable-output.
+ *
+ * @class BasOutput
+ */
+class BaseOutput
+{
+    /**
+     * @var int creates a new image*.png after every run.
+     */
+    protected $imageIndex = 0;
+
+    /**
+     * Provides additional options.
+     *
```

Comment:
Good summary, but add a little bit more explanation for what this method can be used and implemented.
It is especially important to have good documentation for base-classes because there will probably other developers that subclass from a base-class and they are very happy to get as much documentation and explanation as possible.
Example for this method:
`This allows an output to define additional, output-specific options. All outputs are queried for their options so that they can be passed as command-line arguments when running gameoflife.`